fix: removing falco_events prom metric to address OOMing

stefanamaerz opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines in the [CONTRIBUTING.md](https://github.com/falcosecurity/.github/blob/master/CONTRIBUTING.md) file and learn how to compile Falco from source [here](https://falco.org/docs/source).
2. Please label this pull request according to what type of issue you are addressing.
3. If the PR is unfinished while opening it specify a wip in the title before the actual title, for example, "wip: my awesome feature"
-->

**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

/kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

> /kind feature

<!--
Please remove the leading whitespace before the `/kind <>` you uncommented.
-->

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area build

/area config

> /area outputs

> /area tests


<!--
Please remove the leading whitespace before the `/area <>` you uncommented.
-->

**What this PR does / why we need it**:
Addresses an issue where sidekick is ooming due to high cardinality metric collection. Per the conversation here https://github.com/falcosecurity/falcosidekick/issues/1132, rather than toggling the metric on/off, removal of the metric entirely is preferred since the Falco binary itself now exports equivalent data. 

**Which issue(s) this PR fixes**:
https://github.com/falcosecurity/falcosidekick/issues/1132

<!--
Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
If PR is `kind/failing-tests` or `kind/flaky-test`, please post the related issues/tests in a comment and do not use `Fixes`.
-->

Fixes #1132

**Special notes for your reviewer**:
While testing this fix, I submitted 1,000 fake messages to sidekick:

```
for i in $(seq 1 1000); do
  curl -s -X POST http://localhost:2801/ -H "Content-Type: application/json" -d "{
    \"output\": \"Test event from curl\",
    \"priority\": \"Debug\",
    \"rule\": \"Test rule\",
    \"time\": \"2025-11-24T12:00:00Z\",
    \"hostname\": \"testhost$i\",
    \"source\": \"syscalls\",
    \"output_fields\": {
      \"proc.name\": \"curl\",
      \"user.name\": \"tester\"
    },
    \"tags\": [\"test\", \"curl\"]
  }"
done
```

Then outputted the prom scrape to a file:
```
 ~/Desktop/falco-testing/ du -hs *
 12K     without_ falco_events.prom
180K   with_ falco_events old.prom
```

Where `without_ falco_events.prom` is the file generated by sidekick using this branch.

stefanamaerz committed:
fix: removing falco_events prom metric to address OOMing

Signed-off-by: Stefan Maerz <smaerz@fastly.com>

leogr reviewed (commented):
Hi @stefanamaerz 

Thanks for this PR. It seems good to me, but it may worth to cleanup some debries.

For examples:
- Dead regPromLabels global in main.go:  The global variable at lines ~95 and ~112 was only used by the removed code in handlers.go. It's now dead code. Should be removed.
- Dead Prometheus.ExtraLabels / ExtraLabelsList config — The entire config chain is now dead: type fields in types/types.go (lines 364-367), v.SetDefault in config.go (line 550), parsing in config.go (lines 879-881), and docs in config_example.yaml (lines 234-235). Users with this option set will see no effect and no warning.

Furthermore, I believe we have to document this breaking change. Users may have dashboards/alerts on this metric and `prometheus.extralabels` configured.
I worth to add a note in the README.md at least.

stefanamaerz committed:
cleanup

Signed-off-by: Stefan Maerz <smaerz@fastly.com>

stefanamaerz committed:
removing stats_prometheus_test.go since it is now empty

Signed-off-by: Stefan Maerz <smaerz@fastly.com>

stefanamaerz commented on stats_prometheus_test.go:
```diff
@@ -1,28 +1,3 @@
 // SPDX-License-Identifier: MIT OR Apache-2.0
 
 package main
```
https://github.com/falcosecurity/falcosidekick/pull/1286/changes/5cae487f7a4fe79bfc2c554848c4449db476e793 done!

stefanamaerz committed:
docs: remove deprecated prometheus.extralabels from config_example.yaml

Signed-off-by: Stefan Maerz <smaerz@fastly.com>

stefanamaerz commented:
Ok did some updates here: 
* Rebased changes from master
* addressed PR reviewer's feedback, along with some other out of date docs
* Moved to draft

Still todo: I want to test the functionality again, will follow up on this later

poiana commented:
Issues go stale after 90d of inactivity.

Mark the issue as fresh with `/remove-lifecycle stale`.

Stale issues rot after an additional 30d of inactivity and eventually close.

If this issue is safe to close now please do so with `/close`.

Provide feedback via https://github.com/falcosecurity/community.

/lifecycle stale

leogr commented:
Hey @stefanamaerz 

Can you rebase this PR, please? :pray:

poiana commented:
Adding label `do-not-merge/contains-merge-commits` because PR contains merge commits, which are not allowed in this repository.
Use `git rebase` to reapply your commits on top of the target branch. Detailed instructions for doing so can be found [here](https://git.k8s.io/community/contributors/guide/github-workflow.md#4-keep-your-branch-in-sync).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository.
</details>

Issif commented:
IMHO

The disabling should be an opt-in for now, especially because it's a breaking change, and as @leogr said, a lot of people may still use falcosidekick as single endoint to collect metrics about their falco fleets.

The point remains valid, the cardinality reamins an issue, it's a classic with prom world
